diff --git a/browser/components/urlbar/UrlbarController.sys.mjs b/browser/components/urlbar/UrlbarController.sys.mjs
--- a/browser/components/urlbar/UrlbarController.sys.mjs
+++ b/browser/components/urlbar/UrlbarController.sys.mjs
@@ -292,7 +292,6 @@ export class UrlbarController {
     const isMac = AppConstants.platform == "macosx";
     // Handle readline/emacs-style navigation bindings on Mac.
     if (
-      isMac &&
       this.view.isOpen &&
       event.ctrlKey &&
       (event.key == "n" || event.key == "p")
diff --git a/dom/events/ShortcutKeyDefinitions.cpp b/dom/events/ShortcutKeyDefinitions.cpp
--- a/dom/events/ShortcutKeyDefinitions.cpp
+++ b/dom/events/ShortcutKeyDefinitions.cpp
@@ -172,11 +172,11 @@ ShortcutKeyData ShortcutKeys::sInputHand
     /**************************************************************************
      * Common editor commands in <input>.
      **************************************************************************/
-    {u"keypress", nullptr, u"c", u"accel",       u"cmd_copy"},   // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"x", u"accel",       u"cmd_cut"},    // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel",       u"cmd_paste"},  // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel",       u"cmd_undo"},   // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel,shift", u"cmd_redo"},   // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"c", u"control",       u"cmd_copy"},   // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"x", u"control",       u"cmd_cut"},    // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control",       u"cmd_paste"},  // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control",       u"cmd_undo"},   // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control,shift", u"cmd_redo"},   // Win, macOS, Linux, Android, Emacs
 
     {u"keypress", nullptr, u"v", u"accel,shift",     u"cmd_paste"},  // Win, macOS, Linux, Android, Emacs
 // Mac uses Option+Shift+Command+V for Paste and Match Style
@@ -361,11 +361,11 @@ ShortcutKeyData ShortcutKeys::sTextAreaH
     /**************************************************************************
      * Common editor commands in <textarea>.
      **************************************************************************/
-    {u"keypress", nullptr, u"c", u"accel",       u"cmd_copy"},       // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"x", u"accel",       u"cmd_cut"},        // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel",       u"cmd_paste"},      // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel",       u"cmd_undo"},       // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel,shift", u"cmd_redo"},       // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"c", u"control",       u"cmd_copy"},       // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"x", u"control",       u"cmd_cut"},        // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control",       u"cmd_paste"},      // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control",       u"cmd_undo"},       // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control,shift", u"cmd_redo"},       // Win, macOS, Linux, Android, Emacs
 
 #if defined(XP_WIN) || defined(MOZ_WIDGET_GTK) ||\
     defined(USE_EMACS_KEY_BINDINGS)
@@ -556,12 +556,12 @@ ShortcutKeyData ShortcutKeys::sBrowserHa
     /**************************************************************************
      * Common editor commands in non-editable element.
      **************************************************************************/
-    {u"keypress", nullptr, u"c", u"accel",       u"cmd_copy"},              // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"x", u"accel",       u"cmd_cut"},               // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel",       u"cmd_paste"},             // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel,shift", u"cmd_pasteNoFormatting"}, // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel",       u"cmd_undo"},              // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel,shift", u"cmd_redo"},              // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"c", u"control",       u"cmd_copy"},              // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"x", u"control",       u"cmd_cut"},               // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control",       u"cmd_paste"},             // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control,shift", u"cmd_pasteNoFormatting"}, // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control",       u"cmd_undo"},              // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control,shift", u"cmd_redo"},              // Win, macOS, Linux, Android, Emacs
 
 // Mac uses Option+Shift+Command+V for Paste and Match Style
 #if defined(MOZ_WIDGET_COCOA)
@@ -725,12 +725,12 @@ ShortcutKeyData ShortcutKeys::sEditorHan
     /**************************************************************************
      * Common editor commands in HTMLEditor.
      **************************************************************************/
-    {u"keypress", nullptr, u"c", u"accel",           u"cmd_copy"},               // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"x", u"accel",           u"cmd_cut"},                // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel",           u"cmd_paste"},              // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"v", u"accel,shift",     u"cmd_pasteNoFormatting"},  // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel",           u"cmd_undo"},               // Win, macOS, Linux, Android, Emacs
-    {u"keypress", nullptr, u"z", u"accel,shift",     u"cmd_redo"},               // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"c", u"control",           u"cmd_copy"},               // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"x", u"control",           u"cmd_cut"},                // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control",           u"cmd_paste"},              // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"v", u"control,shift",     u"cmd_pasteNoFormatting"},  // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control",           u"cmd_undo"},               // Win, macOS, Linux, Android, Emacs
+    {u"keypress", nullptr, u"z", u"control,shift",     u"cmd_redo"},               // Win, macOS, Linux, Android, Emacs
 
 // Mac uses Option+Shift+Command+V for Paste and Match Style
 #if defined(MOZ_WIDGET_COCOA)
